builder/views.py

```diff
@@ -217,3 +219,17 @@ def new_search(request, draft):
         messages.info(request, f'There are no results for "{term}"')
 
     return redirect(draft.get_builder_search_url(search.slug))
+
+
+@login_required
+@require_http_methods(["POST"])
+@load_draft
+@require_permission
+def delete_search(request, draft, search_slug):
+    search = get_object_or_404(Search, version=draft, slug=search_slug)
+    actions.delete_search(search=search)
+    messages.info(
+        request,
+        f'Search for "{search_slug}" deleted.  Any included codes and their descendants have been maintained.',
```

Comment:
I would change this to something like:

> Search for "{search_slug}" deleted.  Any codes matching "{search_slug}" that do not match any other search term and which are not included in the codelist have been removed

This is not strictly accurate, since it doesn't mention not deleting codes which are descendants of included codes, but I think it gets the balance about right between accuracy and comprehensibility.